### ISSUE TITLE
Support router.push with optional pathname

### ISSUE THREAD

**Issue by KJoobin** [scottrippey/next-router-mock#39]
Hi, i'am using this packages but i have one issue, maybe this not support router push with undefined pathname,
```js
router.push({
  query: {
    five:5
  }
});

console.log(router.pathname) 
// i want return "/PATH/NAME?five=5"
// real "/?five=5"
```

it working in next-router-mock? ( i run add one test-case in your 'pushing URLs should update the route' test but not it's fail)
```js 

        await memoryRouter.push({
          query:{
            five: "5",
            four: "4"
          }
        })

        expect(memoryRouter).toMatchObject({
          asPath: "/one/two/three?four=4&five=5",
          pathname: "/one/two/three",
          query: {
            five: "5",
            four: "4",
          },
        })
```

```
"next": "12.1.0",
"next-router-mock": "^0.6.5",
```

thx

**Comment by scottrippey** [comment]
Currently, this isn't supported by next-router-mock.  I cannot find any [Next.js documentation](https://nextjs.org/docs/api-reference/next/router) that says whether `pathname` is optional.

For now, a workaround would be to do:
```
router.push({
  pathname: router.pathname,
  query: { five: 5 },
});
```

However, this feature could probably be implemented, if you're sure that's how Next.js behaves?  I will try to spin up a test app.

**Comment by scottrippey** [comment]
Here's a test case that shows what you described; calling `router.push({ query: ... })` does in fact allow `pathname` to be undefined.
https://codesandbox.io/s/great-almeida-6g477w?file=/pages/about.js

I'll see if I can fix this.

**Comment by scottrippey** [comment]
Published as `0.6.7`! Enjoy.
